File: app/src/main/java/com/arejas/dashboardofthings/domain/usecases/ActuatorManagementUseCase.java
package com.arejas.dashboardofthings.domain.usecases;

import androidx.lifecycle.LiveData;

import com.arejas.dashboardofthings.domain.entities.database.Actuator;
import com.arejas.dashboardofthings.domain.entities.extended.ActuatorExtended;
import com.arejas.dashboardofthings.domain.entities.extended.SensorExtended;
import com.arejas.dashboardofthings.domain.entities.result.Resource;

import org.jetbrains.annotations.NotNull;

import java.util.List;

public interface ActuatorManagementUseCase extends BaseUseCase {

    public LiveData<Resource<List<ActuatorExtended>>> getListOfActuators();

    public LiveData<Resource<List<ActuatorExtended>>> getListOfActuatorsMainDashboard();

    public LiveData<Resource<List<ActuatorExtended>>> getListOfActuatorsLocated();

    public LiveData<Resource<ActuatorExtended>> getActuator(@NotNull Integer actuatorId);

    public LiveData<Resource> createActuator(@NotNull Actuator actuator);

    public LiveData<Resource> updateActuator(@NotNull Actuator actuator);

    public LiveData<Resource> deleteActuator(@NotNull Actuator actuator);

}

File: app/src/main/java/com/arejas/dashboardofthings/domain/usecases/ActuatorManagementUseCase.kt
package com.arejas.dashboardofthings.domain.usecases

import androidx.lifecycle.LiveData

import com.arejas.dashboardofthings.domain.entities.database.Actuator
import com.arejas.dashboardofthings.domain.entities.extended.ActuatorExtended
import com.arejas.dashboardofthings.domain.entities.extended.SensorExtended
import com.arejas.dashboardofthings.domain.entities.result.Resource

interface ActuatorManagementUseCase : BaseUseCase {

    val listOfActuators: LiveData<Resource<List<ActuatorExtended>>>

    val listOfActuatorsMainDashboard: LiveData<Resource<List<ActuatorExtended>>>

    val listOfActuatorsLocated: LiveData<Resource<List<ActuatorExtended>>>

    fun getActuator(actuatorId: Int): LiveData<Resource<ActuatorExtended>>

    fun createActuator(actuator: Actuator): LiveData<Resource<*>>

    fun updateActuator(actuator: Actuator): LiveData<Resource<*>>

    fun deleteActuator(actuator: Actuator): LiveData<Resource<*>>

}
